src/hidimstat/base_perturbation.py

```diff
@@ -138,14 +148,17 @@ def importance(self, X, y):
             - 'importance': the importance scores for each group.
         """
         self._check_fit(X)
+        random_generator = (
+            None if self.random_state is None else check_random_state(self.random_state)
+        )
```

Comment:
```suggestion
```
Using the attribute `self.random_state` in .predict would also remove the need to define a generator here